src/main/java/com/sandvoxel/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -201,98 +195,119 @@ public boolean canBeRotated() {
         return true;
     }
 
-    public long getPower() {
-        return container.getStoredPower();
+
+
+    public boolean canWork(){
+        if (machineTier == MachineTier.TIER_0){
+            return fuelRemaining > 0;
+        }else  {
+         return container.getStoredPower() >= powerUsage;
+        }
     }
 
+    public boolean cancrush(ItemStack item){
+            return PulverizerRegistry.containsInput(item);
+    }
+
+    public void burnTime(int i)
+    {
+        if (fuelRemaining==0){
+            if (inventory.getStackInSlot(4).getItem() == lastFuelType && cancrush(inventory.getStackInSlot(i))) {
+                fuelRemaining = lastFuelValue;
+
+
+            } else if (inventory.getStackInSlot(4).getItem() != lastFuelType && cancrush( inventory.getStackInSlot(i) ) )
+            {
+                fuelRemaining = net.minecraft.tileentity.TileEntityFurnace.getItemBurnTime(inventory.getStackInSlot(4));
+                lastFuelType = inventory.getStackInSlot(4).getItem();
+                lastFuelValue = fuelRemaining;
+            }
+            fuelTotal = fuelRemaining;
+            inventory.decrStackSize(4, 1);
+            this.markDirty();
+            this.markForUpdate();
+        }
+    }
 
     @Override
     public void update() {
 
-        //container.givePower(1000,false);
-
 
+        //checking for the machine type
         if (machineTier == null)
             machineTier = MachineTier.byMeta(getBlockMetadata());
 
-        if (fuelRemaining == 0 && inventory.getStackInSlot(4) != null && net.minecraft.tileentity.TileEntityFurnace.getItemBurnTime(inventory.getStackInSlot(4)) > 0 && machineTier == MachineTier.TIER_0) {
-            if (inventory.getStackInSlot(0) != null || inventory.getStackInSlot(1) != null) {
-                if (inventory.getStackInSlot(4).getItem() == lastFuelType) {
-                    fuelRemaining = lastFuelValue;
-
-
-                } else {
-                    fuelRemaining = net.minecraft.tileentity.TileEntityFurnace.getItemBurnTime(inventory.getStackInSlot(4));
-                    lastFuelType = inventory.getStackInSlot(4).getItem();
-                    lastFuelValue = fuelRemaining;
+        if (!canWork()) {
+            machineActive = false;
+            this.markForUpdate();
+        }
 
-                }
-                fuelTotal = fuelRemaining;
-                inventory.decrStackSize(4, 1);
-                this.markDirty();
-                this.markForUpdate();
-            }
+        if (ticksRemaining == 0 && inventory.getStackInSlot(0)==null && machineTier != MachineTier.TIER_0){
+            machineActive = false;
         }
 
-        if ((this.container.takePower(powerUsage, true) == powerUsage && machineTier != MachineTier.TIER_0) || fuelRemaining > 0) {
-            if (machineActive && !pulverizerPaused) {
+        if (this.canWork()){
+            if (machineTier == MachineTier.TIER_0){
+                fuelRemaining--;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS00122)